File: app/src/main/java/com/study/android/morden/java/RoomActivity.java
package com.study.android.morden.java;

import android.os.AsyncTask;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.room.Room;

import com.study.android.databinding.ActivityRoomBinding;

public class RoomActivity extends AppCompatActivity {

    private ActivityRoomBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityRoomBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        AppDatabase db = Room.databaseBuilder(this, AppDatabase.class, "todo-db_java")
                .build();

        // UI 갱신
        db.todoDao().getAll().observe(this, todos -> {
            binding.result.setText(todos.toString());
        });

        // 버튼 클릭시 DB에 Insert
        binding.addButton.setOnClickListener(v -> {
            new InsertAyncTask(db.todoDao())
                    .execute(new Todo(binding.result.getText().toString()));
        });

    }

    private static class InsertAyncTask extends AsyncTask<Todo, Void, Void>{
        private TodoDao mTodoDao;

        public InsertAyncTask(TodoDao mTodoDao) {
            this.mTodoDao = mTodoDao;
        }

        @Override
        protected Void doInBackground(Todo... todos) {
            mTodoDao.insert(todos[0]);
            return null;
        }
    }
}

File: app/src/main/java/com/study/android/morden/kotlin/RoomActivity.kt
package com.study.android.morden.kotlin

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.lifecycleScope
import androidx.room.Room
import com.study.android.R
import com.study.android.databinding.ActivityRoomBinding
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class RoomActivity: AppCompatActivity() {

    private lateinit var binding: ActivityRoomBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityRoomBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val db = Room.databaseBuilder(this, AppDatabase::class.java, "todo-db_kotlin")
            .build()


        db.todoDao().getAll().observe(this, Observer { binding.result.text = it.toString() })

        binding.addButton.setOnClickListener {
            lifecycleScope.launch(Dispatchers.IO){
                db.todoDao().insert(Todo(binding.todoEdit.text.toString()))
            }
        }
    }
}